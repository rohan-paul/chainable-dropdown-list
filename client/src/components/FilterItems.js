import React, { Component } from 'react';

class FilterItems extends React.Component {
  render () {
    return (
      <div className="filter-items">
        {this.props.data.map(item => {
          return (<div key={item.name} className="filter-item">{item.name}</div>)
        }
      )}
      </div>

    );
  }
};

export default FilterItems;