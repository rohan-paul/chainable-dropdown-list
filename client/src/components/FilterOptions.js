import React, { Component } from 'react';

class FilterOptions extends React.Component {

  changeOption = (type, e) => {
    let val = e.target.value;
    this.props.changeOptionProp(val, type);
  }

  render () {
    return (
      <div className="filter-options">
        <div className="filter-option">

          <label>Car Make</label>
          <select id="carMake" placeholder="Search by tags" value={this.props.carMake} onChange={this.changeOption.bind(this, 'carMake')}>

          {this.props.carMakeOptions.map(option => (<option key={option} value={option}>{option}</option> ))}
          </select>

          <label>Model</label>
          <select id="model" value={this.props.model} onChange={this.changeOption.bind(this, 'model')}>

          {this.props.modelOptions.map(option => (<option key={option} value={option}>{option}</option> ))}
          </select>

        </div>
      </div>
    );
  }
};

export default FilterOptions;