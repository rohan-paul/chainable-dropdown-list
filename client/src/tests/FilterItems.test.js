import React from 'react';
import expect from 'expect';
import FilterItems from '../components/FilterItems';
import ReactTestUtils from 'react-dom/test-utils'; // ES6


describe('FilterItems', () => {
  it('should exist', () => {
    expect(FilterItems).toBeTruthy();
  });
});

/* describe('FilterItems', function () {
  it('loads without problems', function () {
    require('../components/FilterItems');
  });
}); */

// The below 2 tests are failing - giving " Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined."

describe('FilterItems', function () {
  it('loads without problems', function () {
    var filterItems = ReactTestUtils.renderIntoDocument(<FilterItems
      data={ [1, 2, 3] }
      />);

  expect(filterItems).toBeTruthy();
  });
});

/* describe('FilterItems', function () {

  it('changes without problems', function () {
    var FilterItems =  ReactTestUtils.renderIntoDocument(<FilterItems/>);

    const inputNode = ReactDOM.findDOMNode(FilterItems.refs.input);

    const newValue = 'some text';
    inputNode.value = newValue;
    ReactTestUtils.Simulate.change(inputNode);

    const nameNode = ReactDOM.findDOMNode(FilterItems.refs.name);
    expect(nameNode.textContent).toEqual(newValue);
  });
}); */