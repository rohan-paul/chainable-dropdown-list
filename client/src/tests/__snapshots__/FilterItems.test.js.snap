// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterItems
    data={
        Array [
            1,
            2,
            3,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="filter-item"
/>,
        <div
          className="filter-item"
/>,
        <div
          className="filter-item"
/>,
      ],
      "className": "filter-items",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "filter-item",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "filter-item",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "filter-item",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="filter-item"
/>,
          <div
            className="filter-item"
/>,
          <div
            className="filter-item"
/>,
        ],
        "className": "filter-items",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "filter-item",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "filter-item",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "filter-item",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
